docs/docs/experimental/3-aot.md

```diff
@@ -0,0 +1,141 @@
+---
+sidebar_position: 3
+sidebar_label: AOT compilation
+title: AOT compilation
+---
+## Runtime AOT
+
+<!--
+```java
+//DEPS com.dylibso.chicory:docs-lib:999-SNAPSHOT
+//DEPS com.dylibso.chicory:aot-experimental:999-SNAPSHOT
+
+import com.dylibso.chicory.wasm.Parser;
+import com.dylibso.chicory.runtime.Instance;
+import com.dylibso.chicory.runtime.Machine;
+
+docs.FileOps.copyFromWasmCorpus("count_vowels.rs.wasm", "your.wasm");
+```
+-->
+
+The Ahead-of-Time compiler backend is a drop-in replacement for the interpreter, and it passes 100% of the same
+spec tests that the interpreter already supports.
+
+You can instantiate a module using the AoT by explicitly providing a `MachineFactory`.
+The default `Machine` implementation is the `InterpreterMachine`.
+
+You can opt in to the AoT mode by writing:
+
+```java
+import com.dylibso.chicory.experimental.aot.AotMachine;
+
+var module = Parser.parse(new File("your.wasm"));
+var instance = Instance.builder(module).withMachineFactory(AotMachine::new).build();
+```
+
+after you add the dependency:
+
+```xml
+<dependency>
+  <groupId>com.dylibso.chicory</groupId>
+  <artifactId>aot-experimental</artifactId>
+</dependency>
+```
+
+This will translate every module you instantiate into Java bytecode on-the-fly and in-memory.
+The resulting code is usually expected to evaluate (much)faster and consume less memory.
+
+Please note that compiling and executing AoT modules at runtime requires:
+- an external dependency on [ASM](https://asm.ow2.io/)
+- the usage of runtime reflection
+
+This is usually fine when running on a standard JVM, but it involve some additional configuration when using tools like `native-image`.
+
+## Pre-compiled AOT
+
+You can use the AOT compiler at build-time, by leveraging a [Maven plug-in][aot-maven-plugin] to overcome the usage of reflection and external dependencies of the "Runtime AOT".
```

Comment:
The `aot-maven-plugin` link doesn't work (no suggestion since I'm not sure what you intended)

Reply:
removed